### PR TITLE
feat: Add functionality to disable posts in abandoned negotiations

### PR DESCRIPTION
**Pull request by jungwire**
### Description:

Adds functionality to disable posts in abandoned negotiations

### Checklist:

_Make sure you tick all the boxes bellow if they are true or do not apply:_

- [x] I have performed a self review of my code
- [x] My code follows [Google Java Code style](https://google.github.io/styleguide/javaguide.html)
- [x] I have made my code as simple as possible
- [x] I have added unit tests and the code coverage has not decreased
- [ ] I have updated the documentation in all relevant places

### CODE REVIEW

File: src/main/java/eu/bbmri_eric/negotiator/configuration/state_machine/negotiation/DisablePostsAction.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package eu.bbmri_eric.negotiator.configuration.state_machine.negotiation;
+
+import eu.bbmri_eric.negotiator.service.NegotiationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.statemachine.StateContext;
+import org.springframework.statemachine.action.Action;
+import org.springframework.stereotype.Component;
+
+/** SpringStateMachine action for enabling posts in a Negotiation */
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/configuration/state_machine/negotiation/DisablePostsAction.java:10]: disabling

File: src/main/java/eu/bbmri_eric/negotiator/configuration/state_machine/negotiation/NegotiationStateMachineConfig.java [hunk 2]
```diff
@@ -72,14 +72,25 @@ public void configure(StateMachineTransitionConfigurer<String, String> transitio
...
         .source(NegotiationState.IN_PROGRESS.name())
         .target(NegotiationState.CONCLUDED.name())
-        .event(NegotiationEvent.CONCLUDE.name());
+        .event(NegotiationEvent.CONCLUDE.name())
+        .and()
+        .withExternal()
+        .source(NegotiationState.IN_PROGRESS.name())
+        .target(NegotiationState.ABANDONED.name())
+        .event(NegotiationEvent.ABANDON.name())
+        .action(disablePosts());
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/configuration/state_machine/negotiation/NegotiationStateMachineConfig.java:81]: duplicate code, check few lines above

**Review by RadovanTomik — CHANGES_REQUESTED** [review]
please see comments and add a test in the NegotiationLifecycle tests where you change the state to abandoned and verify that postsEnabled is set to false

File: src/main/java/eu/bbmri_eric/negotiator/configuration/state_machine/negotiation/DisablePostsAction.java (hunk 1 above)
Comment (jungwire) [reply src/main/java/eu/bbmri_eric/negotiator/configuration/state_machine/negotiation/DisablePostsAction.java]: corrected

File: src/main/java/eu/bbmri_eric/negotiator/configuration/state_machine/negotiation/NegotiationStateMachineConfig.java (hunk 2 above)
Comment (jungwire) [reply src/main/java/eu/bbmri_eric/negotiator/configuration/state_machine/negotiation/NegotiationStateMachineConfig.java]: removed duplicated code

**Review by RadovanTomik — APPROVED** [review]

**Pull request merged**